int32 parent
int32[] children

